ddtrace/contrib/grpc/aio_client_interceptor.py

```diff
@@ -0,0 +1,238 @@
+    # type: (Span, aio.AioRpcError) -> None
+    code = to_unicode(rpc_error.code())
+    span.error = 1
+    span._set_str_tag(constants.GRPC_STATUS_CODE_KEY, code)
+    span._set_str_tag(ERROR_MSG, rpc_error.details())
+    span._set_str_tag(ERROR_TYPE, code)
+    span.finish()
+
+
+async def _handle_cancelled_error(call, span):
+    # type: (aio.Call, Span) -> None
+    code = to_unicode(await call.code())
+    span.error = 1
+    span._set_str_tag(constants.GRPC_STATUS_CODE_KEY, code)
+    span._set_str_tag(ERROR_MSG, await call.details())
+    span._set_str_tag(ERROR_TYPE, code)
+    span.finish()
+
+
+class _ClientInterceptor:
+    def __init__(self, pin, host, port):
+        # type: (Pin, str, int) -> None
+        self._pin = pin
+        self._host = host
+        self._port = port
+
+    def _intercept_client_call(self, method_kind, client_call_details):
+        # type: (str, aio.ClientCallDetails) -> Tuple[Span, aio.ClientCallDetails]
+        tracer = self._pin.tracer
+
+        method_as_str = client_call_details.method.decode()
+        span = tracer.trace(
+            "grpc",
+            span_type=SpanTypes.GRPC,
+            service=trace_utils.ext_service(self._pin, config.grpc_aio_client),
+            resource=method_as_str,
+        )
+        span.set_tag(SPAN_MEASURED_KEY)
+
+        utils.set_grpc_method_meta(span, method_as_str, method_kind)
+        utils.set_grpc_client_meta(span, self._host, self._port)
+        span._set_str_tag(constants.GRPC_SPAN_KIND_KEY, constants.GRPC_SPAN_KIND_VALUE_CLIENT)
+
+        sample_rate = config.grpc_aio_client.get_analytics_sample_rate()
+        if sample_rate is not None:
+            span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
+
+        # inject tags from pin
+        if self._pin.tags:
+            span.set_tags(self._pin.tags)
+
+        # propagate distributed tracing headers if available
+        headers = {}
+        if config.grpc_aio_client.distributed_tracing_enabled:
+            HTTPPropagator.inject(span.context, headers)
+
+        metadata = []
+        if client_call_details.metadata is not None:
+            metadata = list(client_call_details.metadata)
+        metadata.extend(headers.items())
+
+        client_call_details = aio.ClientCallDetails(
+            client_call_details.method,
+            client_call_details.timeout,
+            metadata,
+            client_call_details.credentials,
+            client_call_details.wait_for_ready,
+        )
+
+        return span, client_call_details
+
+    async def _wrap_stream_response(
+        self,
+        call,  # type: Union[aio.StreamStreamCall, aio.UnaryStreamCall]
+        span,  # type: Span
+    ):
+        # type: (...) -> ResponseIterableType
+        try:
+            async for response in call:
+                yield response
+            code = await call.code()
+            details = await call.details()
+            call.add_done_callback(_done_callback(span, code, details))
```

Comment:
As far as I tested locally, directly running post-processing here instead of setting the callback worked fine. I used `add_done_callback` because it was also used in existing `ddtrace/contrib/grpc`.
Should I directly call post-processing here?

Reply:
I would use the callback but add a comment about why the implementation has the callback registered after the iterator is finished.

Reply:
Made a comment in 14dc78a.